app/controllers/support_ticket/base_controller.rb

```diff
@@ -16,12 +14,10 @@ def thank_you
 private
 
   def require_support_ticket_data!
-    redirect_to support_ticket_path if session[:support_ticket].blank?
+    redirect_to support_ticket_path unless support_ticket.started?
   end
 
-  def set_support_ticket
-    return unless session[:support_ticket]
-
-    @support_ticket = SupportTicket.new(params: session[:support_ticket])
+  def support_ticket
+    @support_ticket ||= SupportTicket.find_or_create_by!(session_id: session.id.to_s)
```

Comment:
Just one thing to watch out for, we might get some spurious SupportTicket records that get created by the user clicking on the link, but never updated because the user never fills in the form. Shouldn't be many, but just something to keep an eye on

Reply:
I get your point but it's not quite so trivial. There would have to be other changes that would be needed on the first step to check if it's persisted or not as `update` requires a persisted record. The user actually has to hit the start button on the landing page for the record to be persisted which i think is acceptable.

Reply:
OK, we can add some kind of clean-up job to delete old + empty SupportTickets in a separate card if it becomes an issue

Reply:
it's associated with a session. the session sweeper now calls `destroy_all` so will fire AR callbacks which will delete the associated `SupportTicket` which was the old behaviour when a session was lost